airbyte-cdk/python/airbyte_cdk/sources/declarative/schema/json_schema.py

```diff
@@ -4,19 +4,23 @@
 
 import json
 import pkgutil
+import sys
 from dataclasses import InitVar, dataclass, field
 from typing import Any, Mapping, Union
 
-import __main__
 from airbyte_cdk.sources.declarative.interpolation.interpolated_string import InterpolatedString
 from airbyte_cdk.sources.declarative.schema.schema_loader import SchemaLoader
 from airbyte_cdk.sources.declarative.types import Config
 from dataclasses_jsonschema import JsonSchemaMixin
 
 
 def _default_file_path() -> str:
-    main_file = __main__.__file__
-    module = main_file.split("/")[-2].replace("-", "_")
+    # schema files are always in "source_<connector_name>/schemas/<stream_name>.json
+    # the connector's module name can be inferred by looking at the modules loaded and look for the one starting with source_
+    source_modules = [
+        k for k, v in sys.modules.items() if "source_" in k
+    ]  # example: ['source_exchange_rates', 'source_exchange_rates_tutorial.source']
```

Comment:
I'm a little confused by your comment example here. Are you saying that the result of one `sys.modules` call will result in these two items in the map? Or that these are two possible permutations of how the `sys.modules` call will be returned (one called from the `source_exchange_rates` source and one called from the `source_exchange_rates_tutorial` source)?

Reply:
fixed the comment. thanks!